code/modules/mining/drill.dm

```diff
@@ -0,0 +1,338 @@
+//For handling the types of randomized malfunctions
+#define MALF_LASER 1
+#define MALF_SENSOR 2
+#define MALF_CAPACITOR 3
+#define MALF_STRUCTURAL 4
+#define MALF_CALIBRATE 5
+
+//For handling the repair of a completely destroyed drill
+#define METAL_PLACED 1
+#define METAL_SECURED 2
+
+/obj/machinery/drill
+	name = "heavy-duty laser mining drill"
+	desc = "A large scale laser drill. It's able to mine vast amounts of minerals from near-surface ore pockets, however the seismic activity tends to anger local fauna."
+	icon = 'icons/obj/machines/drill.dmi'
+	icon_state = "deep_core_drill"
+	max_integrity = 400
+	density = TRUE
+	anchored = FALSE
+	use_power = NO_POWER_USE
+	layer = ABOVE_ALL_MOB_LAYER
+	armor = list("melee" = 50, "bullet" = 30, "laser" = 30, "energy" = 30, "bomb" = 30, "bio" = 0, "rad" = 0, "fire" = 90, "acid" = 90)
+	component_parts = list()
+
+	var/malfunction
+	var/active = FALSE
+	var/obj/structure/vein/mining
+	var/datum/looping_sound/drill/soundloop
+	var/obj/item/stock_parts/cell/cell
+	var/preload_cell_type = /obj/item/stock_parts/cell
+	var/power_cost = 100
+	var/metal_attached
+
+/obj/machinery/drill/examine(mob/user)
+	. = ..()
+	if(panel_open && component_parts)
+		. += display_parts(user, TRUE)
+	if(cell.charge < power_cost*5)
+		. += "<spawn class='notice'>The low power light is blinking."
+	switch(malfunction)
+		if(4)
```

Comment:
I see this starts at four. Can you add some examine text here for stock part malfunctions?